tools/cabana/videowidget.cc

```diff
@@ -183,13 +183,33 @@ void Slider::sliderChange(QAbstractSlider::SliderChange change) {
   }
 }
 
+bool sortTimelineBasedOnEventPriority(std::tuple<int, int, TimelineType> left, std::tuple<int, int, TimelineType> right){
+  std::map<TimelineType, int> timelinePriority = {
+    {  TimelineType::None, 0 },
+    {  TimelineType::Engaged, 10 },
+    {  TimelineType::AlertInfo, 20 },
+    {  TimelineType::AlertWarning, 30 },
+    {  TimelineType::AlertCritical, 40 },
+    {  TimelineType::UserFlag, 35 }
+  };
+
+  return timelinePriority[std::get<2>(left)] < timelinePriority[std::get<2>(right)];
+}
+
 void Slider::paintEvent(QPaintEvent *ev) {
   QPainter p(this);
   QRect r = rect().adjusted(0, 4, 0, -4);
   p.fillRect(r, timeline_colors[(int)TimelineType::None]);
   double min = minimum() / 1000.0;
   double max = maximum() / 1000.0;
-  for (auto [begin, end, type] : timeline) {
+
+  std::vector<std::tuple<int, int, TimelineType>> sortedTimeline;
```

Comment:
We don't need to sort the same timeline repeatedly in paintEvent. think it's better to sort it in 
```
timer.callOnTimeout([this]() {
    timeline = can->getTimeline();
    update();
  });
```

Reply:
done